jekyll/_posts/2021-11-01-this-month-with-nim.md

```diff
@@ -0,0 +1,163 @@
+    inc self.age
+
+
+# auto define constructor
+let p1 = newPerson("Tony")
+let p2 = newPerson("Steve", 100)
+```
+It has the following features:
+- Member variables with default values
+- Class data constants
+- Definition of `proc`, `method`, `func`, etc... (the only exception being `macro`)
+- Auto inserting `self` as first argument
+- Auto definition of constructor (high performance!)
+- Assistance with constructor definition
+- `pub` modifier instead of `*`
+- Inheritance with `of` modifier
+- Creating distinct type with `distinct` modifier
+- `{.final.}` by default
+- `{.open.}` to allow inheritance
+- `super` keyword for `method`
+- Alias class
+
+More features are planned:
+- `Pick` like TypeScript's `Pick`
+- `let` member variables
+- struct
+- protocol like `interface`
+
+
+## [LibVF.IO](https://libvf.io)
+
+#### Author: [Arthur Rasmusson](https://github.com/arthurrasmusson)
+
+LibVF.IO is commodity GPU multiplexing driven by YAML & VFIO. 
+We wrote an article about it [here](https://telegra.ph/LibVFIO-A-Vendor-Neutral-GPU-Multiplexing-Tool-09-27)
+
+LibVF.IO is made in Nim. :)
+
+
+## [Metaprogramming in Nim #1 Introduction
+](https://youtu.be/2EWLE-IFgGM)
+
+#### Author: [Kiloneie](hhttps://github.com/Kiloneie)
+
+In this video i show and teach Nim's Metaprogramming features/capabilities. Each of it's features are explained and demonstrated in a beginner friendly manner.
+
+## [Nimscripter 1.0.0 - The Rewrittening](https://github.com/beef331/nimscripter)
+
+#### Author: [Jason Beetham](https://github.com/beef331)
+
+Over the past few months I have slowly been chipping at a Nimscripter rewrite and got it to a usable state in October!
+No longer does it rely on silly string serialization or other annoying things, you can now easily expose code to a VM and call it or expose VM code to the compiled code and call it. 
+The following is a simple example:
+```nim
+import nimscripter
+addCallable(someNSModule):
+  proc doThing(a: int): int
+
+var someNum = 10
+proc doOtherThing: bool = true
+exportTo(someNsModule,
+  someNum,
+  doOtherThing)
+const
+  script = """
+proc doThing*(a: int): int = 
+  if doOtherThing():
+    someNum * a
+  else: 0
+"""
+  addins = implNimScriptModule(someNsModule)
+let intr = loadScript(NimScriptFile(script), addins)
+assert intr.invoke(doThing, 2, returnType = int) == 2 * someNum
+someNum = 30
+assert intr.invoke(doThing, 4, returnType = int) == 4 * someNum
+```
+What's that I hear, "Oh it'd be cool if it worked with emscripten"?!
+[Well it does!](https://www.jasonbeetham.com/snake/nicoscripter.html)
+
+## [Turing Complete](https://store.steampowered.com/app/1444480/Turing_Complete/)
+
+#### [Stuffe](https://github.com/Stuffe)
+In this challenging puzzle game, you build a computer from scratch and program it.
+It is a journey through the layers of abstractions of the computer.
+The game begins when you are abducted by aliens who are testing your intelligence, and from here a lighthearted sci-fi plot evolves.
```

Comment:
I feel like this should be at the top of this list :)

Reply:
maybe we can also add info that it's written with godot-nim? it might be a bit unclear from the description

Reply:
Yeah, also can we add an image showing the game's banner? :)

Reply:
> I feel like this should be at the top of this list :)

Yep.

> Maybe we can also add info that it's written with godot-nim?

Yep.

> Yeah, also can we add an image showing the game's banner? :)

Yep.

----

I'll push a change that contain all of these suggestions.